content/creator/sdk7/interactivity/raycasting.md

```diff
@@ -10,17 +10,315 @@ weight: 5
+	- **global**: Ignores the entity's rotation, and faces a direction as if the entity's rotation was 0. This is useful to i.e. always point down.
+	- **global target**: Traces a line between the entity's position and a target global position in the scene. It ignores the entity's rotation. Useful for example to create tower defense games, each tower's turret can point to a pin-pointed coordinate in space.
+	- **target entity**:  Traces a line between the entity's position and the position of a second target entity. It ignores the rotation of either entities.
+
+The following code creates a raycast with a local direction:
+
+```ts
+const myEntity = engine.addEntity()
+Transform.create(myEntity, {
+	position: Vector3.create(4, 1, 4)
+})
+
+raycastSystem.registerLocalDirectionRaycast(
+	myEntity,
+    (raycastResult) => {
+      if (raycastResult.hits.length > 0) {
+        for (const hit of raycastResult.hits) {
+          if (hit.entityId) {
+            // hit some entity
+          }
+        }
+      } else {
+        // no entity hit
+      }
+    },
+    {
+      maxDistance: 30,
+      queryType: RaycastQueryType.RQT_QUERY_ALL,
+      direction: Vector3.Forward(),
+    }
+)
+```
+
+Use the following functions to create raycasts by providing the direction in different ways:
+
+- `raycastSystem.registerLocalDirectionRaycast()`: creates a raycast with a **local** direction. The `direction` field expects a `Vector3` that describes a vector relative to the entity and its rotation (e.g. `Vector3.Forward()` would end up using the entity's transform forward vector)
+- `raycastSystem.registerGlobalDirectionRaycast()`: creates a raycast with a **global** direction. The `direction` field expects a `Vector3` that describes a global rotation.
+- `raycastSystem.registerGlobalTargetRaycast()`: creates a raycast with a direction defined by a **global target** direction. The `target` field expects a `Vector3` that describes a global position in the scene.
+- `raycastSystem.registerTargetEntityRaycast()`: creates a raycast with a direction defined by a **target entity** direction. The `targetEntity` field expects a reference to an entity, this entity's position will be used as the target of the ray.
+
+The following additional fields are *required* when creating a ray with any of the above methods:
+
+- `maxDistance`: _number_ to set the length with which this ray will be traced.
+- `queryType`: _RaycastQueryType_ enum value, to define if the ray will return all hit entities or just the first. The following options are available:
+	- `RaycastQueryType.RQT_QUERY_ALL`: only returns the first hit entity, starting from the origin point.
+	- `RaycastQueryType.RQT_HIT_FIRST`: returns all hit entities, from the origin through to the max distance of the ray.
```

Comment:
Actually no field is really required, since if any property is missing it gets defaulted.

Maybe it can only be mentioned that the specific "direction" property of the chosen raycast type is required.

I don't know if we want to specify or not the default values of a missing property but just in case here they are:
* `maxDistance`: `16`
* `queryType`: `RaycastQueryType.RQT_HIT_FIRST`
* `continuous`: `false`
* `originOffset`: `Vector3.Zero()`
* `collisionMask`: `ColliderLayer.CL_PHYSICS`
* `direction`
    * LocalDirection: `Vector3.Forward()`
    * GlobalDirection: `Vector3.Forward()`
    * GlobalTarget: `Vector3.Zero()`
    * TargetEntity: root entity that's at 0,0,0 of the scene.